packages/react-components/react-dialog/src/components/Dialog/useDialog.ts

```diff
@@ -54,9 +57,18 @@ export const useDialog_unstable = (props: DialogProps): DialogState => {
     requestOpenChange,
   });
 
-  const handlebackdropClick = useEventCallback((event: React.MouseEvent<HTMLDivElement>) => {
+  const { targetDocument } = useFluent_unstable();
+  const disableScroll = useDisableScroll();
+
+  useIsomorphicLayoutEffect(() => {
+    if (open && modalType !== 'non-modal' && targetDocument) {
```

Comment:
would be great to have an e2e test for this feature, where you measure the width of the the window to test if the scrollbar is present

Reply:
actually any test works with unit test you could just check for `overflow: hidden` on the body I guess

Reply:
Great idea, I'll be adding. I'll return this PR as draft for the moment